### ISSUE TITLE
Rejecting a promise with an object gives a blank failure

### ISSUE THREAD

**Issue by jakecraige** [mochajs/mocha#1532]
If a promise is rejected with a normal javascript object the test still fails, but with a blank message. This confused me for the longest time.. It would be great if we could have it at least `JSON.stringify` the object or alert the user that they should return an `Error` object like it does if you resolve with a string.

Test case to reproduce plus show current behavior using RSVP promise library:

```js
var RSVP = require('rsvp');

it('gives helpful message when rejected with string', function() {
  return RSVP.reject('an error message');
});
// => Error: the string "an error message" was thrown, throw an Error :) ... stack

it('gives helpful message when rejected with error', function() {
  return RSVP.reject(new Error('an error message'));
});
// => Error: an error message ... stack

it('gives blank message and no stack when rejected with object', function() {
  return RSVP.reject({});
});
// => nothing
```

**Comment by dasilvacontin** [comment]
Confirmed.

**Comment by dasilvacontin** [comment]
Closed via #1537.

**Comment by boneskull** [comment]
@dasilvacontin @jakecraige 

Sorry I didn't see this earlier.  Please see my comment [here](https://github.com/mochajs/mocha/pull/1537/files#r24287807).
